apps/server/src/modules/user/repo/role.repo.ts

```diff
@@ -0,0 +1,110 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { Role, IPermissionsAndRoles } from '../entity';
+
+interface CacheEntry {
+	[name: string]: { value: Role; validUntil: number };
+}
+
+// TODO: no query handled at the moment
+// TODO: write test and move
+// TODO: replace with generall solution if it is added
+export class RoleCache {
+	cache: CacheEntry;
+
+	name: string;
+
+	// we do not registred changes by calling the old endpoint
+	// also the clear interval should short
+	clearInterval: 60000;
+
+	constructor(name = '') {
+		this.cache = {};
+		this.name = name;
+	}
+
+	get(selector: string): Role | undefined {
+		const entry = this.cache[selector];
+		if (!entry || entry.validUntil < Date.now()) return undefined;
+		return entry.value;
+	}
+
+	add(selector: string, role: Role): void {
+		this.cache[selector] = {
+			value: role,
+			validUntil: Date.now() + this.clearInterval, // TODO: test if it is work
+		};
+	}
+
+	clear(): void {
+		this.cache = {};
+	}
+}
+
+// caching ?
+@Injectable()
+export class RoleRepo {
+	cl1: RoleCache;
+
+	cl2: RoleCache;
```

Comment:
what es the purpose of having a lvl1 and lvl2 cache? the naming makes me think of cpu cache levels, where 1 is smaller but faster than 3, but that doesnt seem to be the case here.